wolfcrypt/src/port/Renesas/renesas_tsip_aes.c

```diff
@@ -151,6 +151,255 @@ int wc_tsip_AesCbcDecrypt(struct Aes* aes, byte* out, const byte* in, word32 sz)
     tsip_hw_unlock();
     return ret;
 }
+/*-------------------------------------------------------------------
 
+    wc_tsip_AesGcmEncrypt
+
+-------------------------------------------------------------------*/
+int wc_tsip_AesGcmEncrypt(
+            Aes*  aes, 
+            byte* out, const byte* in,  word32 sz,
+            byte*       iv,             word32 ivSz,
+            byte*       authTag,        word32 authTagSz,
+            const byte* authIn,         word32 authInSz)
+{
+
+    tsip_gcm_handle_t   hdl;
+    word32              ret;
+    uint32_t            dataLen;  
+    word32              blocks = (sz/ AES_BLOCK_SIZE);
+
+    if ( in == NULL  || out == NULL || aes == NULL ||
+         iv == NULL  || authTag == NULL || authIn == NULL ){   
+        WOLFSSL_MSG("<< wc_tsip_AesGcmEncrypt: Bad Arg");
+      return BAD_FUNC_ARG;
+    }
+ 
+    if (aes->ctx.keySize != 16 && aes->ctx.keySize != 32) {
+        WOLFSSL_MSG("<< wc_tsip_AesGcmEncrypt: illegal key size");
+        return  BAD_FUNC_ARG;
+    }
+
+    if((ret = tsip_hw_lock()) != 0){
+        WOLFSSL_MSG("<< wc_tsip_AesGcmEncrypt: Failed to lock");
+        return ret;
+    }
+
+    /* Initialization step */
+
+    if (aes->ctx.keySize == 16) {
+        ret = R_TSIP_Aes128GcmEncryptInit(&hdl,&aes->ctx.tsip_keyIdx,iv,ivSz);
+    } else{
+        ret = R_TSIP_Aes256GcmEncryptInit(&hdl,&aes->ctx.tsip_keyIdx,iv,ivSz);
+    }
+    if( ret != TSIP_SUCCESS){
+        WOLFSSL_MSG("<< R_TSIP_AesXXXGcmEncryptInit : failed");
+        goto finalize;
+    }
+
+    /* passing additional authentication data step */
+
+    if (aes->ctx.keySize == 16){
+        ret = R_TSIP_Aes128GcmEncryptUpdate(
+            &hdl,(uint8_t*)NULL,(uint8_t*)out,(uint32_t)0,
+            (uint8_t*)authIn,authInSz);
+    }
+    else{
+        ret = R_TSIP_Aes256GcmEncryptUpdate(
+            &hdl,(uint8_t*)NULL,(uint8_t*)out,(uint32_t)0,
+            (uint8_t*)authIn,authInSz);
+    }
+
+    if(ret != TSIP_SUCCESS){
+        WOLFSSL_MSG("R_TSIP_AesxxxGcmEncryptUpdate : failed");
+        goto finalize;
+    }
+  
+    while (ret == TSIP_SUCCESS && blocks--) {
+
+        if (aes->ctx.keySize == 16){
+
+            ret = R_TSIP_Aes128GcmEncryptUpdate(
+                    &hdl,(uint8_t*)in,(uint8_t*)out,(uint32_t)AES_BLOCK_SIZE,  NULL, 0);
+
+        }
+        else{
+
+            ret = R_TSIP_Aes256GcmEncryptUpdate(
+                    &hdl,(uint8_t*)in,(uint8_t*)out,(uint32_t)AES_BLOCK_SIZE,
+                    NULL, 0);
+
+        }
+        in  += AES_BLOCK_SIZE;
+        out += AES_BLOCK_SIZE;
+    }
+
+    if(ret != TSIP_SUCCESS){
+        WOLFSSL_MSG("R_TSIP_AesxxxGcmEncryptUpdate: failed");
+        goto finalize;
+    }
+
+finalize:
```

Comment:
Our coding standards state to avoid using goto unless absolutely necessary.  Can we rearrange this without the use of goto?  Also applies to "exit:" below.

Reply:
Rewrote code without "goto".